etcdmain/gateway.go

```diff
@@ -77,6 +78,23 @@ func newGatewayStartCommand() *cobra.Command {
 	return &cmd
 }
 
+func stripSchema(eps []string) []string {
+	var endpoints []string
+
+	for _, ep := range eps {
+
+		if strings.HasPrefix(ep, "https://") {
```

Comment:
I wanted it to be general and parse the endpoints passed by "--endpoints" param too. I think it could be a common mistake to add "http://" or "https://" to that parameter. 

I could use the url library, but then parse only the endpoints which come from the dns discovery, or use this method for both. I don't think url method would work with correct hostnames like "etcd-node:2379" without further checks.